src/Compilers/Core/MSBuildTaskTests/TestUtilities/DotNetSdkTestBase.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.BuildTasks.UnitTests
+{
+    public abstract partial class DotNetSdkTestBase : TestBase
```

Comment:
I think the tests/framework simulating end-to-end msbuild are great, and will be useful for future work with custom targets. I wonder if the msbuild repo has something similar by now (seems it would be useful there too).

Reply:
The dotnet SDK repo has this kind of test API. The problem is none of these are in a form that's reusable elsewhere. It would indeed be good to create some model that can be reused via a NuGet package.